sound/soc/soc-component.c

```diff
@@ -1213,11 +1213,11 @@ int snd_soc_pcm_component_pm_runtime_get(struct snd_soc_pcm_runtime *rtd,
 	int i;
 
 	for_each_rtd_components(rtd, i, component) {
-		int ret = pm_runtime_get_sync(component->dev);
-		if (ret < 0 && ret != -EACCES) {
-			pm_runtime_put_noidle(component->dev);
+		int ret = pm_runtime_resume_and_get(component->dev);
```

Comment:
@plbossart I think this is not just simplifying but also changing the flow. pm_runtime_resume_and_get() would also fo a put_noidle() in case of -EACCES

Reply:
OK this change is correct but we should clarify that its not just a simplification but also a correction for the incorrect error path flow.

Reply:
done